Disallow toggling encryption in safe mode

pcahyna opened the pull request:
Toggling encryption is (currently) a destructive operation, and this fact is quite nonintuitive and possibly dangerous. Let it be covered by the storage_safe_mode flag, just as with other operations that cause data loss (except for explicit removal).

dwlehman reviewed (approved):
It looks good to me. My only comment is that we should be validating the value of `blivet_output.changed` to make sure it is `False` when we abort due to `storage_safe_mode`. It doesn't necessarily have to be in this patch set, though, I guess.

pcahyna commented:
> It looks good to me. My only comment is that we should be validating the value of `blivet_output.changed` to make sure it is `False` when we abort due to `storage_safe_mode`. It doesn't necessarily have to be in this patch set, though, I guess.

All the new tests have it, don't they? I just pushed a commit to improve formatting of this part, no functional change intended.

pcahyna commented:
Actually, some of tests don't have it. Let me fix it.

pcahyna commented:
I added those checks, which allowed me to find and fix a small error in tests: one pool name was given as `foobar` and not `foo`, here: https://github.com/linux-system-roles/storage/commit/8c091bd4828392216798dcda6575977aa53294a4#diff-ede83c510a6055640001e6b500925559R334 . I also added tests for partition pools.

yizhanglinux commented:
@guazhangRH here is the luks fix MR
